src/components/RepeatableReferenceField.tsx

```diff
@@ -0,0 +1,157 @@
+    });
+  }, [rows, props.sdk.space, referenceKey]);
+
+  // update contentful field value whenever rows data changes
+  useEffect(() => {
+    const sanitizedRows = rows.map((row) => {
+      const sanitizedRow = {};
+      sanitizedRow[textKey] = row[textKey];
+      sanitizedRow[referenceKey] = row[referenceKey];
+      return sanitizedRow;
+    });
+    props.sdk.field.setValue(sanitizedRows);
+  }, [rows, props.sdk.field, referenceKey, textKey]);
+
+  // open entry selection dialog and append selected entries to the end of our list
+  const onAddButtonClicked = () => {
+    const options = {};
+    if (contentTypes) {
+      options.contentTypes = contentTypes;
+    }
+    props.sdk.dialogs.selectMultipleEntries(options)
+      .then((selectedRows) => {
+        setRows([
+          ...rows,
+          ...selectedRows.map((row) => {
+            const rowData = {
+              key: `${row.sys.id}-${Math.floor(Math.random() * 100000)}`
+            };
+            rowData[textKey] = '';
+            rowData[referenceKey] = row.sys.id;
+            return rowData;
+          })
+        ]);
+        props.sdk.field.setValue(rows);
+      })
+      .catch(() => { /* do nothing */ });
+  };
+
+  // update ingredients with new amount
+  const onTextChanged = (e) => {
+    const rowIndex = e.target.dataset.index;
+    const updatedRows = [...rows];
+    updatedRows[rowIndex][textKey] = e.target.value;
+    setRows(updatedRows);
+  }
+
+  // remove ingredient from list
+  const onDeleteButtonClicked = (e) => {
+    let actualTarget = e.target;
+    while (!actualTarget.dataset.index || actualTarget.id === 'root') {
+      actualTarget = actualTarget.parentNode;
+    }
+    if (actualTarget.id === 'root') {
+      return;
+    }
+    const rowIndex = parseInt(actualTarget.dataset.index);
+    const updatedRows = rows.filter((_, index) => index !== rowIndex);
+    setRows(updatedRows);
+  }
+
+  return <section>
+      <div>
+        <Table>
+          <TableBody>
+              {rows.map((row, index) => {
+                return <TableRow key={row.key}>
+                  <TableCell>
+                      <TextInput 
+                        value={row[textKey]}
+                        placeholder={textLabel}
+                        data-index={index}
+                        onChange={onTextChanged}>
+                      </TextInput>
+                  </TableCell>
+                  <TableCell style={{ width: '200px' }}>
+                    {row.name ? row.name :
+                      <SkeletonContainer svgHeight="20">
+                        <SkeletonBodyText numberOfLines="1"></SkeletonBodyText>
+                      </SkeletonContainer>
+                    }
+                  </TableCell>
+                  <TableCell>
+                      <EditorToolbarButton 
+                        icon="Delete"
+                        data-index={index}
+                        onClick={onDeleteButtonClicked}>
```

Comment:
After inspecting the logic of `onDeleteButtonClicked`, I think this could be simplified if you were to instead do something like 
```js
onClick={() => onDeleteButtonClicked(row)}
```

Then in the function itself you should be able to more quickly do something like 
```js
rows.filter(r => r.sys.id !== passedInRow.sys.id)
```

Reply:
Nice one! Done: 04bff6b

Reply:
Nice, yeah I think that simplifies things quite a bit.